File: web/src/types/order.ts
export type OrderStatus =
  | "pending"
  | "confirmed"
  | "processing"
  | "shipped"
  | "delivered"
  | "cancelled"

export type PaymentMethod = "card" | "cod" | "bank_transfer"
export type PaymentStatus = "unpaid" | "paid" | "failed" | "refunded"

export interface OrderItem {
  productId: string
  slug: string
  name: string
  image: string
  price: number
  quantity: number
}

export interface ShippingAddress {
  fullName: string
  phone: string
  line1: string
  line2?: string
  city: string
  state: string
  postalCode: string
  country: string
}

export interface OrderStatusEvent {
  status: OrderStatus
  at: string
}

export interface Order {
  id: string
  number: string
  status: OrderStatus
  items: OrderItem[]
  subtotal: number
  shipping: number
  tax: number
  discount: number
  total: number
  paymentMethod: PaymentMethod
  paymentStatus?: PaymentStatus
  paymentReference?: string
  paidAt?: string
  couponCode?: string
  shippingAddress: ShippingAddress
  notes?: string
  statusHistory?: OrderStatusEvent[]
  createdAt: string
  deliveredAt?: string
  eta?: string
}

export interface Coupon {
  id: string
  code: string
  type: "percent" | "fixed"
  value: number
  minOrder: number
  description: string
  expiresAt: string
}

File: backend/app/schemas/order.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import OrderStatus, PaymentMethod


class OrderItemRequest(BaseModel):
    product_id: UUID
    quantity: int = Field(ge=1)


class ShippingAddressRequest(BaseModel):
    full_name: str
    phone: str
    line1: str
    line2: str | None = None
    city: str
    state: str
    postal_code: str
    country: str


class OrderCreateRequest(BaseModel):
    items: list[OrderItemRequest] = Field(min_length=1)
    payment_method: PaymentMethod
    coupon_code: str | None = None
    shipping_address: ShippingAddressRequest
    notes: str | None = Field(default=None, max_length=500)


class OrderItemResponse(BaseModel):
    product_id: UUID
    slug: str
    name: str
    image: str | None = None
    price: float
    quantity: int


class ShippingAddressResponse(BaseModel):
    full_name: str
    phone: str
    line1: str
    line2: str | None = None
    city: str
    state: str
    postal_code: str
    country: str


class OrderResponse(BaseModel):
    id: UUID
    number: str
    status: OrderStatus
    items: list[OrderItemResponse]
    subtotal: float
    shipping: float
    tax: float
    discount: float
    total: float
    payment_method: PaymentMethod
    coupon_code: str | None
    shipping_address: ShippingAddressResponse
    notes: str | None
    eta: str | None
    delivered_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OrderStatusUpdate(BaseModel):
    status: OrderStatus
